### PR TITLE
Improve some docs in std.traits

### PR DESCRIPTION
**Pull request by JackStouffer**

### CODE REVIEW

File: std/traits.d [hunk 1]
```diff
@@ -5489,6 +5489,27 @@ enum bool isBoolean(T) = is(BooleanTypeOf!T) && !isAggregateType!T;
...
 
+///
+@safe unittest
+{
+    static assert(isIntegral!byte);
+    static assert(isIntegral!short);
+    static assert(isIntegral!int);
+    static assert(isIntegral!long);
+
+    static assert(!isIntegral!double);
```
Comment (andralex) [inline std/traits.d]: Add tests for `bool` and `char` please.

Comment (andralex) [inline std/traits.d]: For compactness you may group all true asserts together and all false asserts together:

```d
static assert(isIntegral!byte && isIntegral!short && ...);
static assert(!isIntegral!double && ...);
```

In fact if you exhaustively enumerate all cases for `true` there's no need for example for `false`.

File: std/traits.d [hunk 2]
```diff
@@ -5489,6 +5489,27 @@ enum bool isBoolean(T) = is(BooleanTypeOf!T) && !isAggregateType!T;
...
+
+    // types which act as integral values do not pass
+    struct S
+    {
+        int val;
+
```
Comment (andralex) [inline std/traits.d]: why would there ever be an empty line in a 4-line struct... cc @wilzbach

File: std/traits.d [hunk 3]
```diff
@@ -5525,6 +5539,30 @@ enum bool isFloatingPoint(T) = __traits(isFloating, T) && !(is(Unqual!T == cfloa
                                                             is(Unqual!T == idouble) ||
                                                             is(Unqual!T == ireal));
 
+///
+@safe unittest
+{
+    static assert(isFloatingPoint!float);
+    static assert(isFloatingPoint!double);
+    static assert(isFloatingPoint!real);
```
Comment (andralex) [inline std/traits.d]: add a qualifier somewhere to clarify it works with qualifiers as well

**Comment by JackStouffer** [comment]
@andralex These tests are purely for documentation. The more comprehensive tests already exist

**Comment by andralex** [comment]
I understand, but really "how about char and bool?" and "does it work with qualifiers?" are the first that come to any user's mind!

**Comment by JackStouffer** [comment]
@andralex Integrated your suggestions while still trying to keep it as clear as possible.

**Pull request merged**